File: increasing-triplet-subsequence/increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        n = len(nums)
        first = float('inf')
        second = float('inf')

        for i in range(0, n):
            if(nums[i] <= first):
                first = nums[i]
            elif(nums[i] <= second):
                second = nums[i]
            else:
                return True
       
        return False

File: 334-increasing-triplet-subsequence/increasing-triplet-subsequence.cpp
#define ll long long
#define vi vector<int>
#define vll vector<ll>
#define vs vector<string>
#define mpi map<int, int>
#define mps map<string, int>
#define pb push_back
#define all(x) (x).begin(), (x).end()

const int MOD = 1e9 + 7;

class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int first = INT_MAX;
        int second = INT_MAX;

        for (auto it : nums) {
            if (it <= first) {
                first = it;
            } else if (it <= second) {
                second = it;
            } else {
                return true;
            }
        }

        return false;
    }
};